unified_planning/plans/sequential_plan.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2021 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import networkx as nx # type: ignore
+import unified_planning as up
+import unified_planning.plans as plans
+from unified_planning.model import FNode, InstantaneousAction
+from unified_planning.walkers import Substituter, Simplifier, FreeVarsExtractor
+from typing import Callable, Dict, Optional, Set, List, cast
+
+
+class SequentialPlan(plans.plan.Plan):
+    '''Represents a sequential plan.'''
+    def __init__(self, actions: List['plans.plan.ActionInstance']):
+        self._actions = actions
+
+    def __repr__(self) -> str:
+        return str(self._actions)
+
+    def __eq__(self, oth: object) -> bool:
+        if isinstance(oth, SequentialPlan) and len(self._actions) == len(oth._actions):
+            for ai, oth_ai in zip(self._actions, oth._actions):
+                if ai.action != oth_ai.action or ai.actual_parameters != oth_ai.actual_parameters:
+                    return False
+            return True
+        else:
+            return False
+
+    @property
+    def actions(self) -> List['plans.plan.ActionInstance']:
+        '''Returns the sequence of action instances.'''
+        return self._actions
+
+    def replace_action_instances(self, replace_function: Callable[['plans.plan.ActionInstance'], 'plans.plan.ActionInstance']) -> 'up.plans.plan.Plan':
+        return SequentialPlan([replace_function(ai) for ai in self._actions])
+
+    def to_partial_order_plan(self, _env: Optional['up.Environment'] = None) -> 'up.plans.partial_order_plan.PartialOrderPlan':
+        env = up.environment.get_env(_env)
```

Comment:
I think that any kind of plan shall contain an environment. Can we put it inside the `Plan` object?

@alvalentini what do you think?

Reply:
Yes, I think it make sense. Now the environment is part of the actions, but I agree that we should put it also inside the `Plan`. Do you think that then we should add a check that the environments are the same?